ls1intum/Artemis
# `Development`: Split continuous integration service to lower coupling between server services

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises

### Motivation and Context
This PR aims to lower the coupling between server services and to reduce the risk of having circular dependencies for those. It creates smaller services and moves some functionality from services into repositories. While the refactoring should not lead to any issues if all server tests and the e2e tests pass, we should still test that the functionality is still working correctly.

The PR also removes some old workarounds which are not necessary any more such as loading build logs from the CI server. (we store them in the database since quite some time)

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Students

This PR refactors multiple aspects of Artemis that need to be tested:

*Static Code Analysis:*
1. Log in to Artemis as an instructor.
2. Create a  programming exercise for a course exercise with SCA enabled (should work with Java and C on TS1 and with Java and Swift on TS2)
3. Configure the grading of the exercise to grade some of the SCA checks. 
4. Log in a s a student and submit a solution containing bad code (e.g. add an unused import) and make sure the result actually reflects the SCA findings.
5. Check the same thing for exam exercises.

*Triggering Builds:*
1. Create a programming exercise and participate as a student.
2. Find the build plan links on the exercise overview page of the instructor and go to the student's build plan.
3. Manually trigger the build from the UI (e.g. from the page containing all participations for the exercise).
4. Check that the builds are triggered in the CI system.
5. Check that solution and template repository builds are triggered when submitting a change to the tests repository.

*Build Results and Build Logs:*
1. Create a programming exercise and participate as a student.
2. Submit a solution that earns you more than 0 points (e.g. add the bubbleSort implementation for the standard Java exercise: https://github.com/ls1intum/Artemis/blob/develop/src/test/cypress/fixtures/exercise/programming/all_successful/BubbleSort.txt)
3. Make sure that the result received contains a message for the failed test cases as well as logs hinting to what went wrong and correctly distinguishes between failed and passed tests.
6. Submit a failed build (e.g. remove a bracket) and check that build logs are displayed instead of successful and failed tests.
7. Check the same thing for exam exercises.

### Review Progress






#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage
no changes except deletion of unused code

### Screenshots
no changes
